src/sas/sasgui/guiframe/local_perspectives/plotting/parameters_panel_slicer.py

```diff
@@ -519,14 +534,36 @@ def send_to_fitting(self, fit=FIT_OPTIONS[0], file_list=None):
                                     self.data_panel.bt_import.GetId())
             wx.PostEvent(self.data_panel, evt)
 
-    def on_auto_save_checked(self, evt=None):
+    def on_checbox_checked(self, evt=None):
         """
-        Enable/Disable auto append when checkbox is checked
+        All wx.checkbox events go to the same method so we have get the event
+        object to determine which box was checked and thus which code to
+        execute.  For this class the two checkbox events are used to:
+        
+        a) Enable/Disable auto append
+        b) togle between +/-q or abs q (|q|) slicer averaging/plotting
         :param evt: Event
-        """
-        self.append_name.Enable(self.auto_save.IsChecked())
-        self.path.Enable(self.auto_save.IsChecked())
-        self.fitting_options.Enable(self.auto_save.IsChecked())
+        
+        """
+        _cb = evt.GetEventObject()
+        _cb_not_found = True
+        _param_index = next((i for i, v in enumerate(self.parameters)
+                                     if v[0] == "abs q"), None)
+        if (_param_index != None):
+            if _cb == self.abs_box:
+                _cb_not_found = False
+                (self.parameters[_param_index])[1] = _cb
```

Comment:
It is odd that checking the checkbox for abs q will set the control associated in abs q to that checkbox.  Shouldn't the proper control have been put into the parameter list when the control was created?